index_batcher.go

```diff
@@ -0,0 +1,164 @@
+package bleve
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+// NOTE XXX As this approach uses carefully orchestrated interactions
```

Comment:
Good point - added a description of the protocol for synchronization between the batch loop and the operation methods along with a bit of explanation as to why.